presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -270,6 +271,27 @@ public long getRegionSizeInBytes(int positionOffset, int length)
         return sizeInBytes;
     }
 
+    @Override
+    public long getPositionsSizeInBytes(boolean[] positions)
+    {
+        checkValidPositions(positions, positionCount);
+
+        boolean[] seen = new boolean[dictionary.getPositionCount()];
+        boolean[] used = new boolean[dictionary.getPositionCount()];
+        for (int i = 0; i < positions.length; i++) {
+            if (positions[i]) {
+                int position = getId(i);
+                if (!seen[position]) {
+                    if (!dictionary.isNull(position)) {
+                        used[position] = true;
+                    }
+                    seen[position] = true;
```

Comment:
I don't think you need the `seen` variable here. I also don't think you need the `isNull` check here at all. I also don't see why the null check in `getRegionSizeInBytes` is necessary.

In addition, you forgot to change `DictionaryBlock.getRegionSizeInBytes`

Reply:
I just assumed that `dictionary.isNull` is more expensive than checking a boolean variable. But I didn't verify that assumption at all. If it's not expected to be expensive then I'll remove it. I didn't change `getRegionSizeInBytes` because `getPositionsSizeInBytes` is not always cheaper. It might make sense for both of them to be recursive so they can be chosen to suit different use cases.